include/s3select_functions.h

```diff
@@ -643,6 +646,23 @@ struct _fn_substr : public base_function{
     
 };
 
+struct _fn_lower : public base_function {
+
+    value v_str;
+    char buff[4096];
+
+    bool operator()(std::vector<base_statement*> * args,variable * result)
+    {
+        std::vector<base_statement*>::iterator iter = args->begin();
+        base_statement* str =  *iter;
+        v_str = str->eval();
+        strcpy(buff,v_str.str());
+        boost::algorithm::to_lower(buff);
```

Comment:
starting with boost seems like a good choice, we use boost::algorithm methods in many places in RGW